### PR TITLE
Enable parameter injection for @TempDir and run as initialization interceptor for fields

### PR DESCRIPTION
**Pull request by leonard84**
* Enable parameter injection for @TempDir.
* Prior to this commit @TempDir values where not available to other field initializers,
now they can be used to initialize other fields.
* Fixed exception when configured baseDir was not existing,
now @TempDir will create the baseDir directory if it is missing.

### CODE REVIEW

**Comment by AndreasTu** [comment]
@leonard84 I guess this waits on #1692 because it contains duplicated code.
Shall I wait for the other PR or have a look at it nonetheless.

**Comment by leonard84** [comment]
@AndreasTu yes it is based on the other PR, but you can have a look at the last 4 commits, as those are unique to this.

File: spock-core/src/main/java/org/spockframework/util/IThrowableBiConsumer.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2009 the original author or authors.
```
Comment (AndreasTu) [inline spock-core/src/main/java/org/spockframework/util/IThrowableBiConsumer.java:2]: Wrong year

File: spock-core/src/main/java/org/spockframework/util/IThrowableBiConsumer.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.spockframework.util;
+
+/**
+ * A BiConsumer that may throw a checked exception.
+ *
+ * @author Peter Niederwieser
```
Comment (AndreasTu) [inline spock-core/src/main/java/org/spockframework/util/IThrowableBiConsumer.java:22]: Wrong author.

**Review by AndreasTu — APPROVED** [review]
Just two small nitpicks.
Thanks for the changes!

**Pull request merged**